nat-lab/tests/test_batching.py

```diff
@@ -0,0 +1,278 @@
+        cnodes = zip(env.clients, env.nodes, [c.connection for c in env.connections])
+
+        # the plan is to misalign the peers for better batcing observation. WIthout misalignment
+        # peers will form a meshnet almost immediately, already achieving the desired batching effect.
+        for client in env.clients:
+            await client.stop_device()
+
+        tasks = []
+        for name in container_names:
+            task = asyncio.create_task(
+                capture_histograms(
+                    name,
+                    capture_duration,
+                    [
+                        PacketFilter("icmp", lambda p: p.haslayer(ICMP)),
+                        PacketFilter(
+                            "udp", lambda p: p.haslayer(IP) and p.haslayer(UDP)
+                        ),
+                        PacketFilter(
+                            "non-derp udp",
+                            lambda p: p.haslayer(IP)
+                            and p.haslayer(UDP)
+                            and not (p[UDP].sport == 3479 or p[UDP].dport == 3479),
+                        ),
+                    ],
+                )
+            )
+            tasks.append(task)
+
+        api = env.api
+
+        # misalign the peers by sleeping some before starting each node again
+        for client, node, _ in cnodes:
+            await asyncio.sleep(3)
+            await client.simple_start()
+            await client.set_meshmap(api.get_meshmap(node.id))
+
+        await asyncio.gather(
+            alpha_client.wait_for_state_peer(
+                beta.public_key,
+                [State.Connected],
+                [PathType.Direct],
+            ),
+            alpha_client.wait_for_state_peer(
+                gamma.public_key,
+                [State.Connected],
+                [PathType.Direct],
+            ),
+            beta_client.wait_for_state_peer(
+                alpha.public_key,
+                [State.Connected],
+                [PathType.Direct],
+            ),
+            beta_client.wait_for_state_peer(
+                gamma.public_key,
+                [State.Connected],
+                [PathType.Direct],
+            ),
+            gamma_client.wait_for_state_peer(
+                alpha.public_key,
+                [State.Connected],
+                [PathType.Direct],
+            ),
+            gamma_client.wait_for_state_peer(
+                beta.public_key,
+                [State.Connected],
+                [PathType.Direct],
+            ),
+        )
+
+        futures = await asyncio.gather(*tasks)
+        for name, histograms in zip(container_names, futures):
+            for filter_name, hs in histograms:
+                title = f"{name}-filter({filter_name})-id({request.node.callspec.id})"
+                draw_histogram(title, hs, max_height=12)
+                silence_streak = longest_zero_streak(hs)
+                print(f"Longest silence for {title} is {silence_streak}\n")
```

Comment:
It counts the empty buckets in a row, but buckets may vary in size, is it really ok? Wouldn't it be better to just get the largest difference between consecutive packets during histogram generation?

Reply:
Ok, it seems that it is set to 1, but I guess if we depend on that it would be good to make it just hardcoded, or at least make it default or sth, searching for the place where it is decided might be painful

Reply:
That's a very valid point. To be honest this metric was not really that much useful so instead of fixing it I just removed it entirely. Your comment on finding largest consecutive difference was good though.